### ISSUE TITLE
Add ionic app as one of the stencil starter template

### ISSUE THREAD

**Issue by beenotung** [stencil-community/proposals#2]
## Package Request

**Name**: stencil-ionic-app-starter

**Package**: `@stencil-community/stencil-ionic-app-starter`

### Description

I propose to add [this starter template](https://github.com/beenotung/stencil-ionic-starter) as an option of starter template when creating a project with `npm init stencil`, probably select this template with `npm init stencil ionic-app`.

This starter template is configured with the current version of ionic (v6.1.14) and @stencil/core (v2.17.3) with tabs layout and routes.

**Comment by rwaskiewicz** [comment]
Hey @beenotung 👋 

Thanks for the proposal! 
A few things:
1. Taking a look at this repo, I'm currently unable to run/build the project after cloning it locally, then running `npm i && npm start`. I've included a screenshot below:

![Screen Shot 2022-09-06 at 9 55 10 AM](https://user-images.githubusercontent.com/1930213/188653796-e7ddc042-cfb2-44ff-ad42-2b60690fefbb.png)

2. For us to accept this into the community, the following steps would be required 
    1. Moving this package to the community `@stencil-community` 
    2. Adding this as an option to  https://github.com/ionic-team/create-stencil
Is that something you would be OK with? We can help you with the transfer aspect and review the PR to add it to https://github.com/ionic-team/create-stencil, but we want to make sure you're OK with that first before we start anything

**Comment by sean-perkins** [comment]
@rwaskiewicz the build error shown is an issue with Ionic Framework (we forgot to remove a custom path alias, since those aren't swapped out in the compiler right now).

That has been fixed here: https://github.com/ionic-team/ionic-framework/pull/25898 and will be available in next week's release (6.2.7).

**Comment by beenotung** [comment]
I'm happy to move the package to the community org, and to update the create-stencil cli package to link to the template :smile:

**Comment by beenotung** [comment]
I just tried to perform the repository transfer, but it shows "You don’t have the permission to create public repositories on stencil-community".

Maybe someone in the org can fork this repo under the org?

**Comment by rwaskiewicz** [comment]
@beenotung I just invited you to the `stencil-community` organization. Can you accept that invite and try again to transfer the repo?

**Comment by beenotung** [comment]
I've just accepted the invite and transferred the starter repo into https://github.com/stencil-community/stencil-ionic-starter

Maybe we shall also find a way to allow community members to update the hosted demo in the future?

**Comment by rwaskiewicz** [comment]
@beenotung 

> I've just accepted the invite and transferred the starter repo into https://github.com/stencil-community/stencil-ionic-starter

Thanks!

> Maybe we shall also find a way to allow community members to update the hosted demo in the future?

I can see if we can get a tweet out to announce adding this to the community organization!

---

Would you be up for opening a PR in the [create-stencil](https://github.com/ionic-team/create-stencil) repo to add this to the list of options when people run `npm init stencil`? It would entail adding a new option to the [starters.ts](https://github.com/ionic-team/create-stencil/blob/56dd902f84bf790a29d46ea622d28a91309c4bb8/src/starters.ts) file that looks very similar to [the options for the existing app starter](https://github.com/ionic-team/create-stencil/blob/56dd902f84bf790a29d46ea622d28a91309c4bb8/src/starters.ts#L50-L56), something like:
```
  {
    name: 'stencil-ionic-starter',
    repo: 'stencil-community/stencil-ionic-starter',
    description: 'Ionic PWA starter with tabs layout and routes ',
    docs: 'https://github.com/stencil-community/stencil-ionic-starter',
    isCommunity: true,
  },
```
and [adding this starter to the README ](https://github.com/ionic-team/create-stencil/blob/main/README.md?plain=1#L47-L50).

[This line of the starter](https://github.com/stencil-community/stencil-ionic-starter/blob/master/package.json#L2) will also need to be updated so that when folks are asked the name of their project, the value in `package.json` gets updated properly. Today, create-stencil replaces all instances of the string "stencil-starter-project-name" with the project name that the user supplies, so the `package.json` of the starter would look something like:
```diff

{
-  "name": "stencil-ionic-starter",
+  "name": "stencil-starter-project-name",
  "private": true,
```

**Comment by beenotung** [comment]
@rwaskiewicz Thanks for the detail guides, I've just submitted a PR to the create-stencil repo[1] :rocket: 

[1] https://github.com/ionic-team/create-stencil/pull/93

**Comment by rwaskiewicz** [comment]
Thanks @beenotung - I'm having some trouble getting the PR to work properly. It's failing to download the repo at the moment, I'll try to circle back to this later today

**Comment by rwaskiewicz** [comment]
Ah, I see the issue. @beenotung can you change the primary branch of the template from `master` to `main`? I forgot about that detail in my original instructions

**Comment by beenotung** [comment]
@rwaskiewicz I've created a branch "main" based on "master", but it seems I don't have the permission to change the primary branch because I cannot see the "Settings" tab on the repository page

**Comment by rwaskiewicz** [comment]
@beenotung I updated your permissions, you should be able to see the "Settings" tab now.

I've also swapped out the default branch from "master" to "main" on the template repository

**Comment by rwaskiewicz** [comment]
@beenotung I've published [v3.1.0 of the create-stencil CLI](https://github.com/ionic-team/create-stencil/releases/tag/v3.1.0) 🎉  Thanks so much for your contribution!

Let me know if there's anything else we need to do before we close this proposal out

**Comment by rwaskiewicz** [comment]
I'm going to close this proposal as accepted. If there's anything else needed, please feel free to comment on this one!
